File: Linear Shoal/Array_I/Q2. Shuffle the Array/Q2_ShuffletheArray.java
// Solution class
class Solution {

    // Method to shuffle the array
    public int[] shuffle(int[] nums, int n) {

        // Store total length of the array
        int len = nums.length;

        // Loop through the second half of the array
        // This packs two numbers into one position
        for (int i = n; i < len; i++) {

            // Store nums[i] and nums[i - n] together using multiplication
            nums[i] = (nums[i] * 1024) + nums[i - n];
        }

        // Index to place shuffled elements
        int index = 0;

        // Loop again through the packed values
        for (int i = n; i < len; i++, index += 2) {

            // Extract the smaller number using modulo
            nums[index] = nums[i] % 1024;

            // Extract the bigger number using division
            nums[index + 1] = nums[i] / 1024;
        }

        // Return the shuffled array
        return nums;
    }
}

File: Linear Shoal/Array_I/Q2. Shuffle the Array/Q2_ShuffletheArray.py
# Solution class
class Solution(object):

    # Function to shuffle the array
    def shuffle(self, nums, n):

        # Create empty list to store result
        l = []

        # Store first n elements (x part)
        l1 = nums[:n]

        # Store remaining elements (y part)
        l2 = nums[n:]

        # Loop n times
        for i in range(n):

            # Add element from first list
            l.append(l1[i])

            # Add element from second list
            l.append(l2[i])

        # Return the shuffled list
        return l
